Dnn.AdminExperience/Dnn.PersonaBar.Extensions/admin/personaBar/Dnn.CssEditor/scripts/CssEditor.js

```diff
@@ -97,26 +98,82 @@ define([
             if (params.settings.isHost) {
                 getPortals();
             }
-            getStyleSheet();
+            
+            initCssEditor();
 
-            cssEditor = codeEditor.init($panel.find('textarea'), { mode: 'css' });
             var panelHeight = $('#CssEditor-panel').height();
             if (panelHeight > 400) {
-                cssEditor.setSize("100%", panelHeight - 330);
+                $('#monaco-editor').height(panelHeight - 400);
             }
 
             if (typeof callback === 'function') {
                 callback();
             }
 
-            cssEditor.on("blur", function(cm) {
+            $('#monaco-editor').on("blur", function(cm) {
                 cm.save();
                 return true;
             });
 
             viewModel.portal.subscribe(portalChanged);
         };
 
+        var initCssEditor = function() {
+            var monacoEditorLoaderScript = document.createElement('script');
+            monacoEditorLoaderScript.type = 'text/javascript';
+            monacoEditorLoaderScript.src = '/Resources/Shared/components/MonacoEditor/loader.js';
+            document.body.appendChild(monacoEditorLoaderScript);
+
+            require.config({ paths: { 'vs': '/Resources/Shared/components/MonacoEditor' }});
+            require(['vs/editor/editor.main'], function(monaco) {
+        
+                self.MonacoEnvironment = {
+                    getWorkerUrl: function (moduleId, label) {
+                        if (label === 'typescript' || label === 'javascript') {
+                        return `data:text/javascript;charset=utf-8,${encodeURIComponent(`
+                            importScripts('${process.env.ASSET_PATH}/typescript.worker.js');`
+                        )}`;
+                        }
+        
+                    return `data:text/javascript;charset=utf-8,${encodeURIComponent(`
+                        importScripts('${process.env.ASSET_PATH}/editor.worker.js');`
+                    )}`;
+                    }
+                }
+
+                cssEditor = monaco.editor;
+                cssContent = cssEditor.createModel("", "css");
+                initMonacoEditor();
+                getStyleSheet();
+            });
+
+        }
+
+        var initMonacoEditor = function() {
+            var theme = "vs-light";
+            if (window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches) {
+                theme = "vs-dark";
+            }
+            window.matchMedia('(prefers-color-scheme: dark)').addEventListener('change', e => {
+                theme = e.matches ? "vs-dark" : "vs-light";
+            });
```

Comment:
@ahoefling regarding your question about switching themes, it looks like the current support is making use of [the OS dark/light theme preference](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme).

Reply:
If it follows the Operating System theme that is probably good enough for now

Reply:
Correct @bdukes - it detects the user's preference and themes monaco-editor accordingly.